janusgraph-core/src/main/java/org/janusgraph/diskstorage/indexing/IndexMutation.java

```diff
@@ -68,22 +78,24 @@ public void resetDelete() {
         isDeleted=false;
     }
 
-    public static final Function<IndexEntry,String> ENTRY2FIELD_FCT = new Function<IndexEntry, String>() {
-        @Nullable
-        @Override
-        public String apply(final IndexEntry indexEntry) {
-            return indexEntry.field;
-        }
-    };
+    private boolean isCollection(String field) {
+        KeyInformation keyInformation = storeRetriever.get(field);
+        return keyInformation != null && keyInformation.getCardinality() != Cardinality.SINGLE;
+    }
+
+    private final Function<IndexEntry, AbstractMap.SimpleEntry<String, Object>> entryConversionFunction =
+            indexEntry -> isCollection(Objects.requireNonNull(indexEntry).field) ?
```

Comment:
Also, could you explain why are you executing `Objects.requireNonNull(indexEntry).field` ?
As for me it is unnecessary because both `Objects.requireNonNull(indexEntry).field` and `indexEntry.field` produces the same `NullPointerException`.